test/data/EnumerableMap.ts

```diff
@@ -0,0 +1,415 @@
+              instance['at(uint256)'](0),
+            ).to.be.revertedWithCustomError(
+              instance,
+              'EnumerableMap__IndexOutOfBounds',
+            );
+          });
+        });
+      });
+
+      describe('#contains(address)', () => {
+        it('returns true if value has been added', async () => {
+          await instance['set(address,address)'](addressOne, addressFour);
+          await instance['set(address,address)'](addressTwo, addressFive);
+          await instance['set(address,address)'](addressThree, addressSix);
+
+          expect(await instance['contains(address)'](addressOne)).to.be.true;
+          expect(await instance['contains(address)'](addressTwo)).to.be.true;
+          expect(await instance['contains(address)'](addressThree)).to.be.true;
+        });
+
+        it('returns false if value has not been added', async () => {
+          expect(await instance['contains(address)'](addressFour)).to.be.false;
+        });
+      });
+
+      describe('#length()', () => {
+        it('returns length of enumerable map', async () => {
+          expect(await instance['length()']()).to.equal(0);
+
+          await instance['set(address,address)'](addressOne, addressFour);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['set(address,address)'](addressTwo, addressFive);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['set(address,address)'](addressThree, addressSix);
+          expect(await instance['length()']()).to.equal(3);
+
+          await instance['remove(address)'](addressThree);
+          expect(await instance['length()']()).to.equal(2);
+
+          await instance['remove(address)'](addressTwo);
+          expect(await instance['length()']()).to.equal(1);
+
+          await instance['remove(address)'](addressOne);
+          expect(await instance['length()']()).to.equal(0);
+        });
+      });
+
+      describe('#get(address)', () => {
+        it('returns address stored at key', async () => {
+          await instance['set(address,address)'](addressOne, addressFour);
+          await instance['set(address,address)'](addressTwo, addressFive);
+          await instance['set(address,address)'](addressThree, addressSix);
+
+          expect(await instance['get(address)'](addressOne)).to.eq(addressFour);
+          expect(await instance['get(address)'](addressTwo)).to.eq(addressFive);
+          expect(await instance['get(address)'](addressThree)).to.eq(
+            addressSix,
+          );
+        });
+
+        describe('reverts if', () => {
+          it('key does not exist', async () => {
+            await expect(
+              instance['get(address)'](addressOne),
+            ).to.be.revertedWithCustomError(
+              instance,
+              'EnumerableMap__NonExistentKey',
+            );
+          });
+        });
+      });
+
+      describe('#set(address,address)', () => {
```

Comment:
We should include a test which demonstrates that calling `set` for a key that already exists overwrites the value.

```js
it('overwrites the address value at address key', async () => {
  await instance['set(address,address)'](addressOne, addressTwo);
  await instance['set(address,address)'](addressOne, addressThree);

  expect(await instance['get(address)'](addressOne)).to.eq(
    addressThree,
  );
});
```